app/controllers/barbeque/sns_subscriptions_controller.rb

```diff
@@ -50,6 +61,8 @@ def destroy
 
   def fetch_sns_topic_arns
     sns_client.list_topics.topics.map(&:topic_arn)
+  rescue Aws::Errors::MissingCredentialsError, Aws::SNS::Errors::AuthorizationError
```

Comment:
I think these exceptions should be reported as 5xx error because sns:ListTopics permission is required in Barbeque